src/par_iter/from_par_iter.rs

```diff
@@ -22,6 +33,27 @@ impl<PAR_ITER, T> FromParallelIterator<PAR_ITER> for Vec<T>
     }
 }
 
+/// Collect items from a parallel iterator into a vecdeque.
+impl<PAR_ITER, T> FromParallelIterator<PAR_ITER> for VecDeque<T>
+    where Vec<T>: FromParallelIterator<PAR_ITER>,
+          T: Send,
+{
+    fn from_par_iter(par_iter: PAR_ITER) -> Self {
+        Vec::from_par_iter(par_iter).into()
+    }
+}
+
+/// Collect items from a parallel iterator into a binaryheap.
+/// The heap-ordering is calculated serially after all items are collected.
```

Comment:
Nit: I'd prefer not to *specify* that it is computed *serially*, though I think we probably should guarantee some amount of left-to-right-ness.